materialize-snowflake/sqlgen.go

```diff
@@ -134,8 +134,12 @@ ALTER TABLE {{ $.Table.Identifier }} ADD COLUMN
 	{{- end }};
 {{ end }}
 
+-- Templated query which joins keys from the load table with the target table, and returns values. It
+-- deliberately skips the trailing semi-colon as these queries are composed with a UNION ALL.
+
 {{ define "loadQuery" }}
-	SELECT {{ $.Table.Binding }}, {{ $.Table.Identifier }}.{{ $.Table.Document.Identifier }}
+{{ if $.Table.Document -}}
```

Comment:
Is it only when delta updates are enabled that there is no Document? 🤔

Reply:
It's only possibly to explicitly exclude the `flow_document` if delta updates are enabled, otherwise it is required: https://github.com/estuary/connectors/blob/88fe5895ef5df5c90e7cd8930f2e3bcf19113abf/materialize-sql/validate.go#L71-L76

And by default even with delta updates enabled `flow_document` is recommended, so it will almost always still be there. But we do actually have at least one production task running that is configured like this, for another materialization.